src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,141 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for constructing Vietoris-Rips complexes and computing the persistence diagrams from them.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
+        threshold=float('inf'),
+        input_type='point cloud',
+        num_collapses=True,
+        homology_coeff_field=11,
+        expand_extra_dimension = True,
+        n_jobs=None,
+    ):
+        """
+        Constructor for the RipsPersistence class.
+
+        Parameters:
+            homology_dimensions (int or list of int): The returned persistence diagrams dimension(s).
+                Short circuit the use of :class:`~gudhi.representations.preprocessing.DimensionSelector` when only one
+                dimension matters (in other words, when `homology_dimensions` is an int).
+            threshold (float): Rips maximal edge length value. Default is +Inf.
+            input_type (str): Can be 'point cloud' when inputs are point clouds, or 'lower distance matrix', when
+                inputs are lower triangular distance matrix (can be full square, but the upper part of the distance
+                matrix will not be considered). Default is 'point cloud'.
+            num_collapses (bool or int): Specify if :func:`~gudhi.SimplexTree.collapse_edges` is performed. When
+                `num_collapses` is an integer, it specifies the number of :func:`~gudhi.SimplexTree.collapse_edges`
+                iterations to perform on the SimplexTree. Default is True, which means "automatic" (a relatively good
+                enough number of iterations is choosen).
```

Comment:
Having something that can be int or bool but where True means something different from 1 doesn't sound right to me.

Reply:
Ok, let's stick with an int (default to 1 is good enough for the moment) on [86b0a00](https://github.com/GUDHI/gudhi-devel/pull/691/commits/86b0a00a1679ddaff6c0f4485a8d4cb2cdb6e8d5).
We could have some magic numbers in the future API like:
* '-1' for automatic number of collapse detection
* '-2' collapse until it is no more collapsable
* ...